Add optional Pod Disruption Budget for each potential Pega tier (#317)
pegasystems/pega-helm-charts: charts/pega/README.md

### Ensure System Availability during Voluntary Disruptions by Using a Kubernetes Pod Disruption Budget (PDB)
To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", 
Pega allows you to enable a unique Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).

You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true` in your values.yaml file.  By default, this value is
set to `false.`  You must also specify at least one of the following parameters to configure the Pod Disruption Budget.  Keep in mind that these parameters
are mutually exclusive, and thus only one may be set.  You may provide values that are expressed as percentages (% of pods) or integers (# of pods).  

Parameter             | Description    | Default value
---                   | ---       | ---
`pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`

> `pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%`

Reviewer: sorry but I think that this is still off...how about?>
The minumum number or percentage of total pods in the tier that can be unavailable at one time; if the number or percentage reaches your minimum, the deployment will not allow you to bring down additional pods until one of the currently unavailable pods is deployed and healthy.

The maximum number or percentage of total pods in the tier that can be unavailable at one time; if the number or percentage reaches your maximum, the deployment will not allow you to bring down additional pods until one of the currently unavailable pods is deployed and healthy.

Reviewer 2: I like the inclusion of percentage, but the suggested description isn't quite technically accurate. Maybe something like this?

> The minimum number or percentage of pods in the tier that must be available. If this minimum is reached, the Kubernetes deployment will not bring down additional pods for voluntary disruptions until more are available and healthy.

> The maximum number or percentage of pods in the tier that can be unavailable. If this maximum is reached, the Kubernetes deployment will not bring down additional pods for voluntary disruptions until more are available and healthy.

`hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
`hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false

### Setting a Pod Disruption Budget (PDB)

> You can configure a Kubernetes `PodDisruptionBudget`on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 

Reviewer 2: ```suggestion
You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 
```

> You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 

Reviewer: for line 483, can we add a context and not rely on the user to put together performance and a "disruption budget" as an equivalence?
To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", Pega allows you to enable a Kubernetes `PodDisruptionBudget`. For more details, see the Kubernetes...

[Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).

> The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.

Reviewer: I think you should state that clients configure this in their deployment values.yaml file and it's disabled ('false') by default. Also, mention that you must use comments to disable the other setting.

Finally, the values.yaml files each have a 50% value commented out, not an integer so I'm automatically confused.

> The following parameters are mutually exclusive.  You may specifiy either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.   

Reviewer 2: ```suggestion
The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
```

Parameter             | Description    | Default value
---                   | ---       | ---
`pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`

> `pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | Disabled when `pdb.minAvailable` is specified

Reviewer: this isn't a value, it's information about usage. Should this be in the description? also, make it active>>Disabled when you specify a `pdb.minAvailable` value.

Should this mention percentage if the commented out values are percentages?

`hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
`hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false

### Ensure System Availability during Voluntary Disruptions by Using a Kubernetes Pod Disruption Budget (PDB)

> To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", 

Reviewer 2: ```suggestion
To limit the number of Pods running your Pega Platform application that can go down for planned disruptions, 
```

> Pega allows you to enable a unique Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).

Reviewer 2: ```suggestion
Pega allows you to enable a Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
```

You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true` in your values.yaml file.  By default, this value is

> set to `false.`  You must also specify at least one of the following parameters to configure the Pod Disruption Budget.  Keep in mind that these parameters

Reviewer 2: ```suggestion
set to `false.`  You must also specify exactly one of the following parameters to configure the Pod Disruption Budget.  These parameters
```

are mutually exclusive, and thus only one may be set.  You may provide values that are expressed as percentages (% of pods) or integers (# of pods).  

Parameter             | Description    | Default value
---                   | ---       | ---
`pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`

> `pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%`

Reviewer 2: ```suggestion
`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%` (disabled by default)
```
